A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/CDU/A320_Neo_CDU_IdentPage.js

```diff
@@ -16,6 +16,73 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+const months = ["JAN", "FEB", "MAR", "APR", "MAY", "JUN", "JUL", "AUG", "SEP", "OCT", "NOV", "DEC"];
+const monthLength = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
+
+function findNewMonthIndex(index) {
+    if (index === 0) {
+        return 11;
+    } else {
+        return index - 1;
+    }
+}
+
+function lessThan10(num) {
+    if (num < 10) {
+        return `0${num}`;
+    } else {
+        return num;
+    }
+}
+
+function calculateActiveDate(date) {
+    if (date.length === 13) {
+        const startMonth = date.slice(0, 3);
+        const startDay = date.slice(3, 5);
+
+        const endMonth = date.slice(5, 8);
+        const endDay = date.slice(8, 10);
+
+        return `${startDay}${startMonth}-${endDay}${endMonth}`;
+    } else {
+        return date;
+    }
+}
+
+function calculateSecDate(date) {
+    if (date.length === 13) {
+        const primStartMonth = date.slice(0, 3);
+        const primStartDay = date.slice(3, 5);
+
+        const primStartMonthIndex = months.findIndex((item) => item === primStartMonth);
+
+        if (primStartMonthIndex === -1) {
+            return "ERR";
+        }
+
+        let newEndMonth = primStartMonth;
+        let newEndDay = primStartDay - 1;
+
+        let newStartDay = newEndDay - 27;
+        let newStartMonth = primStartMonth;
+
+        if (newEndDay === 0) {
+            newEndMonth = months[findNewMonthIndex(primStartMonthIndex)];
+            newEndDay = monthLength[findNewMonthIndex(primStartMonthIndex)];
+        }
+
+        if (newStartDay <= 0) {
+            newStartMonth = months[findNewMonthIndex(primStartMonthIndex)];
+            newStartDay = monthLength[findNewMonthIndex(primStartMonthIndex)] + newStartDay;
+        }
+
+        return `${lessThan10(newStartDay)}${newStartMonth}-${lessThan10(newEndDay)}${newEndMonth}`;
+    } else {
+        return "ERR";
+    }
+
+}
+
```

Comment:
Can you move this to a helper class in the utils  folder?

Reply:
I in all honesty am not sure how